add style guide (#894)
pulumi/pulumi-hugo: themes/default/style-guide.md

# Style Guide
This document defines some general styles we adhere to in the docs.

## Words are important

- Instead of _guys_ try _folks_, _yall_, or _everyone_
- Instead of _crazy_ try _wild_

> - Instead of _click_ use _select_

Author: Dave and I had aligned on using select specifically, can you say more about why we want choose to be an option as well?

> - Instead of _click_ use _select_

Reviewer: Just as a way to keep prose from becoming repetitive. They mean pretty much the same thing and we tend to use them both conversationally, so IMO it's generally okay to use either one. (Though "select" seems fine as a default.)

- Instead of _dummy_ use _placeholder_

> - Avoid using gendered language

Author: id rather this not be a specific example in this situation and is up with the direct-swap list. i can add an example here of `man-hours` though if that would be helpful

- Avoid using violent language (e.g., _kill_)
- Avoid pop-culture references

## Links

- Link text should be descriptive and be understandable outside of the context that it is in.
- Avoid link text like _here_, _click here_, _see here_

## Notes and Warning

- Use notes to communicate important information

> - Try and limit the use of notes within a single page

Author: @cnunciato i mean on a single page. ideally if you have 3 paragraphs and 10 notes that can be pretty difficult to read.

> - Try and limit the use of notes within a single page

Reviewer: Yep, ok, makes sense. So would this work? (Otherwise, it reads a bit like "don't use notes", where really we mean, "don't use several notes".)

```suggestion
- Try and limit the number of notes within a single page
```

> - Use info for general information, and warnings for more important information

Author: @cnunciato would you be up for writing up the situations in which you should use a warning and adding a commit?

> - Use info for general information, and warnings for more important information

Reviewer: Yep, happy to.

> - Use info for general information, and warnings for more important information

Reviewer: @susanev Just pushed a commit -- let me know what you think!

> - Use info for general information, and warnings for more important information

Reviewer: Ah wait, I think I committed over your rename -- will fix.

# Style Guide
This document defines some general styles we adhere to in the docs.

## Words are important

- Instead of _guys_ try _folks_, _yall_, or _everyone_
- Instead of _crazy_ try _wild_
- Instead of _click_ use _select_
- Instead of _dummy_ use _placeholder_
- Avoid using gendered language
- Avoid using violent language (e.g., _kill_)
- Avoid pop-culture references

## Headings

- You should be able to scan the headings of a document and get a good sense of the page
- Every page should have exactly one h1
- Never skip a heading level, e.g., if your previous heading level was a 2, the next heading can be a 2 or a 3, but not a 4 or 5

## Links

- Link text should be descriptive and be understandable outside of the context that it is in.

> - Avoid link text like _here_, _click here_, _see here_

Reviewer: 💯

## Notes and Warning

- Use notes to communicate important information
- Try and limit the use of notes within a single page
- Use info for general information, and warnings for more important information

### Example

```
{{% notes type="info" %}}

> This is a info note.

Reviewer: ```suggestion
This is an info note.
```

{{% /notes %}}

{{% notes type="warning" %}}

> This is a warning  note.

Reviewer: ```suggestion
This is a warning note.
```

{{% /notes %}}
```

> ## Quotes

Reviewer: ```suggestion
## Blockquotes
```

> - Use quotes only when you are quoting a person.

Reviewer: I think it's okay to use blockquotes for citing content in general (e.g., from other texts), so I'd broaden this slightly:

```suggestion
- Use blockquotes only when you are quoting content from another source.
```
